lgc/disassembler/Disassembler.cpp

```diff
@@ -336,7 +384,7 @@ void ObjDisassembler::gatherRelocs(ELFSectionRef sectionRef, std::vector<object:
 // @param symbols : Sorted array of symbols in the section
 // @param relocs : Sorted array of relocs in the section
 void ObjDisassembler::tryDisassembleSection(ELFSectionRef sectionRef, unsigned sectType, unsigned sectFlags,
-                                            bool outputting, ArrayRef<SymbolInfoTy> symbols,
+                                            bool outputting, SymbolPool &symbols,
                                             ArrayRef<object::RelocationRef> relocs) {
```

Comment:
This function is very long. Could we split it up into smaller functions before refactoring?

Reply:
Done in https://github.com/GPUOpen-Drivers/llpc/pull/1889.